src/vivarium/framework/lookup/table.py

```diff
@@ -42,20 +44,20 @@ class LookupTable(ABC):
 
     table_number: int
     """Unique identifier of the table."""
-    data: Union[ScalarValue, pd.DataFrame, List[ScalarValue], Tuple[ScalarValue]]
+    data: ScalarValue | pd.DataFrame | list[ScalarValue] | tuple[ScalarValue]
     """The data from which to build the interpolation."""
-    population_view_builder: Callable = None
+    population_view_builder: Callable[..., PopulationView] | None = None
     """Callable to get a population view to be used by the lookup table."""
-    key_columns: Union[List[str], Tuple[str]] = ()
+    key_columns: list[str] | tuple[str, ...] = ()
     """Column names to be used as categorical parameters in Interpolation
     to select between interpolation functions."""
-    parameter_columns: Union[List[str], Tuple] = ()
+    parameter_columns: list[str] | tuple[str, ...] = ()
     """Column names to be used as continuous parameters in Interpolation."""
-    value_columns: Union[List[str], Tuple[str]] = ()
+    value_columns: list[str] | tuple[str, ...] = ()
     """Names of value columns to be interpolated over."""
     interpolation_order: int = 0
     """Order of interpolation. Used to decide interpolation strategy."""
-    clock: Callable = None
+    clock: Callable[[], ClockTime] | None = None
```

Comment:
`clock` is only not None if it's an `InterpolatedTable`. It should probably be an attribute of `InterpolatedTable`, not `LookupTable`. This will remove the need for some null checks later.